### ISSUE TITLE
Configuring with disabled monitoring still builds it (and fails, if there are missing libraries)

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#497]
Originally reported of Google Code with ID 441.

```
<b>What steps will reproduce the problem?</b>
./configure --disable-monitoring
make
...
/usr/bin/ghc --make \
	  -O -Wall -Werror -isrc -fwarn-monomorphism-restriction -fwarn-tabs -fwarn-incomplete-uni-patterns \
	  -DNO_CURL -DPARALLEL3 -DNO_REGEX_PCRE \
	  -osuf mon-collector.o -hisuf mon-collector.hi \
	    src/mon-collector

Same effect leaving it to autodetection.
Also monitoring requires "Crypto" (eg) but just checks for "attoparsec"

Thanks,

Guido

```
Originally added on 2013-04-30 13:17:34 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
I confirm that mon-collector is compiled anyway, and I'll fix it.

But Crypto is not required by monitoring. It is actually required by confd, that, in turn, is required by monitoring.
Do we already have a way to deal with this kind of "cascading" dependencies?
```
Originally added on 2013-04-30 13:36:00 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
commit 5d453688e67788af492a87c9828cdfdbf94c1de4
Author: Michele Tartara <mtartara@google.com>
Date:   Tue Apr 30 14:01:42 2013 +0000

    Add confd as a dependency of monitoring
    
    Some data collectors require confd data information. Therefore, monitoring is
    dependent on confd, but this was not checked.
    
    This commit introduces such a check and partially fixes Issue 441.
    
    Signed-off-by: Michele Tartara <mtartara@google.com>
    Reviewed-by: Helga Velroyen <helgav@google.com>

commit b43e836fb048b73dd28f2172ea797ded226251b3
Author: Michele Tartara <mtartara@google.com>
Date:   Tue Apr 30 14:15:44 2013 +0000

    Disable compiling mon-collector if monitoring is disabled
    
    Mon-collector kept being compiled even if monitoring was disabled during
    configure.
    
    Fixes Issue 441.
    
    Signed-off-by: Michele Tartara <mtartara@google.com>
    Reviewed-by: Guido Trotter <ultrotter@google.com>
```
Originally added on 2013-04-30 15:55:44 +0000 UTC.

Mentioned Issue with GoogleCode ID 441 has been migrated to GitHub with ID #497.
Changed State: Fixed

**Comment by Ganeti-Issues-Migrator** [comment]
```
Marking 2.7 bugs fixed, as 2.7.0~rc1 was released, and should be ready to be tested in production environments.
```
Originally added on 2013-05-09 05:42:49 +0000 UTC.

Changed State: Released
